pkg/controller/metrics.go

```diff
@@ -0,0 +1,67 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
+
+import (
+	"sync"
+
+	"k8s.io/component-base/metrics"
+	"k8s.io/component-base/metrics/legacyregistry"
+)
+
+const controllerNamespace = "controller"
+
+var (
+	Reconciliations = metrics.NewCounterVec(
+		&metrics.CounterOpts{
+			Namespace:      controllerNamespace,
+			Name:           "reconcile_total",
+			Help:           "Number of times a controller performed a sync of its resources",
```

Comment:
Quibble but given where we seem to be incrementing this counter, this is not the number of times a controller performed a sync, its the number of time a controller started or attempted to perform a sync.